pkg/async/notifications/implementations/sandbox_processor.go

```diff
@@ -0,0 +1,59 @@
+package implementations
+
+import (
+	"context"
+	"time"
+
+	"github.com/flyteorg/flyteadmin/pkg/async"
+	"github.com/flyteorg/flyteadmin/pkg/async/notifications/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/golang/protobuf/proto"
+)
+
+type SandboxProcessor struct {
+	email interfaces.Emailer
+}
+
+func (p *SandboxProcessor) StartProcessing() {
+	for {
+		logger.Warningf(context.Background(), "Starting SandBox notifications processor")
+		err := p.run()
+		logger.Errorf(context.Background(), "error with running processor err: [%v] ", err)
+		time.Sleep(async.RetryDelay)
+	}
+}
+
+func (p *SandboxProcessor) run() error {
+	var emailMessage admin.EmailMessage
+
+	for {
+		select {
+		case msg := <-msgChan:
+			err := proto.Unmarshal(msg, &emailMessage)
+			if err != nil {
+				logger.Errorf(context.Background(), "error with unmarshalling message [%v]", err)
+				return err
+			}
+
+			err = p.email.SendEmail(context.Background(), emailMessage)
+			if err != nil {
+				logger.Errorf(context.Background(), "error with sendemail message [%v] ", err)
+				return err
+			}
+		default:
+			logger.Debugf(context.Background(), "no message to process")
+		}
+	}
+}
+
+func (p *SandboxProcessor) StopProcessing() error {
+	logger.Debug(context.Background(), "call to sandbox stop processing.")
+	return nil
+}
+
+func NewSandboxProcessor(emailer interfaces.Emailer) interfaces.Processor {
```

Comment:
Could we add a test for the sandbox processor that uses the default noop emailer?

Reply:
Yes, I will try it